packages/cache/src/internal/cacheHttpClient.ts

```diff
@@ -113,6 +114,21 @@ export async function getCacheEntry(
   const cacheResult = response.result
   const cacheDownloadUrl = cacheResult?.archiveLocation
   if (!cacheDownloadUrl) {
+    // List cache for primary key only if cache miss occurs
+    const resource = `caches?key=${encodeURIComponent(keys[0])}`
+    const response = await httpClient.getJson<ArtifactCacheList>(getCacheApiUrl(resource))
+    if(response.statusCode === 204) {
+      const cacheListResult = response.result
+      const totalCount = cacheListResult?.totalCount
+      if(totalCount && totalCount > 0) {
+        core.info(`Cache miss occurred on the cache key '${keys[0]}' and version '${version} but there is ${totalCount} existing version of the cache for this key. More info on versioning can be found here: https://github.com/actions/cache#cache-version`)
```

Comment:
We should also mention the current scope if possible. Is it available in Context?

Reply:
Yes